LibVLCSharp/Shared/Core.cs

```diff
@@ -85,7 +150,46 @@ static IntPtr PreloadNativeLibrary(string nativeLibrariesPath, string libraryNam
                 return IntPtr.Zero;
             }
 
-            return Native.LoadLibrary(libraryPath);// TODO: cross-platform load
+            if(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return Native.LoadLibrary(libraryPath);
+            }
+            else if(RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return LinuxNative.dlopen(libraryPath, LinuxNative.RTLD_LAZY);
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        internal static T GetLibraryFunction<T>(IntPtr libraryHandle, string functionName)
```

Comment:
I think these definitions belong to `MarshalUtils`

Reply:
then `PreloadNativeLibrary` too.

I think the loading architecture could be done in a separate class if you want, but maybe it's another PR? Maybe that other PR could be done before this one?
I've mixed a few things here, the linux .so loading and the real implementation of the logging API.
I'm just rebasing for now, will see in the future what to do with that.